mino/minows/address.go

```diff
@@ -0,0 +1,156 @@
+package minows
+
+import (
+	"go.dedis.ch/dela"
+	"strings"
+
+	"github.com/libp2p/go-libp2p/core/peer"
+	ma "github.com/multiformats/go-multiaddr"
+	"go.dedis.ch/dela/mino"
+	"go.dedis.ch/dela/serde"
+	"golang.org/x/xerrors"
+)
+
+const protocolP2P = "/p2p/"
+
+// address represents a publicly reachable network address that can be used
+// to establish communication with a remote player through libp2p and,
+// therefore, must have both `location` and `identity` components.
+// - implements mino.Address
+type address struct {
+	location ma.Multiaddr
+	identity peer.ID
+}
+
+// newAddress creates a new address from a publicly reachable location with a
+// peer identity.
+func newAddress(location ma.Multiaddr, identity peer.ID) (address, error) {
+	if location == nil || identity.String() == "" {
+		return address{}, xerrors.New("address must have location and identity")
+	}
```

Comment:
`peer.ID` has a `Validate()` function on it which returns an error already, we could split this in two.

```suggestion
	if location == nil {
		return address{}, xerrors.New("address must have a location")
	}
	
	if err := identity.Validate(); err {
		return address{}, xerrors.Errorf("address must have a valid identity: %v", err)
	}
```